modules/configuration/src/test/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtension.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.ignite.internal.configuration.asm.ConfigurationAsmGenerator;
+import org.apache.ignite.internal.configuration.hocon.HoconConverter;
+import org.apache.ignite.internal.configuration.tree.ConfigurationSource;
+import org.apache.ignite.internal.configuration.tree.InnerNode;
+import org.apache.ignite.internal.configuration.util.ConfigurationUtil;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.platform.commons.support.AnnotationSupport;
+import org.junit.platform.commons.support.HierarchyTraversalMode;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.ignite.configuration.annotation.ConfigurationType.LOCAL;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * JUnit extension to inject configuration instances into test classes.
+ *
+ * @see InjectConfiguration
+ */
+public class ConfigurationExtension implements BeforeEachCallback, AfterEachCallback, ParameterResolver {
+    /** Key to store {@link ConfigurationAsmGenerator} in {@link ExtensionContext.Store}. */
+    private static final Object CGEN_KEY = new Object();
+
+    /** {@inheritDoc} */
+    @Override public void beforeEach(ExtensionContext context) throws Exception {
+        ConfigurationAsmGenerator cgen = new ConfigurationAsmGenerator();
+
+        context.getStore(ExtensionContext.Namespace.GLOBAL).put(CGEN_KEY, cgen);
+
+        Object testInstance = context.getRequiredTestInstance();
+
+        for (Field field : getMatchingFields(testInstance.getClass())) {
+            field.setAccessible(true);
+
+            field.set(testInstance, cfgValue(field.getType(), field.getAnnotation(InjectConfiguration.class), cgen));
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public void afterEach(ExtensionContext context) throws Exception {
+        context.getStore(ExtensionContext.Namespace.GLOBAL).remove(CGEN_KEY);
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean supportsParameter(
+        ParameterContext parameterContext, ExtensionContext extensionContext
+    ) throws ParameterResolutionException {
+        return parameterContext.isAnnotated(InjectConfiguration.class)
+            && supportType(parameterContext.getParameter().getType());
+    }
+
+    /** {@inheritDoc} */
+    @Override public Object resolveParameter(
+        ParameterContext parameterContext, ExtensionContext extensionContext
+    ) throws ParameterResolutionException {
+        Parameter parameter = parameterContext.getParameter();
+
+        var cgen = (ConfigurationAsmGenerator)extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(CGEN_KEY);
```

Comment:
there's a `get(Object, Class)` method that you can use here

Reply:
also, this line is too long

Reply:
done